src/Microsoft.AspNetCore.StaticFiles/ResponseCacheFilter.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.StaticFiles
+{
+    internal class ResponseCacheFilter : IResponseCacheFilter
+    {
+        /// <summary>
+        /// Apply the given cacheProfile to the <see cref="HttpContext"/>
+        /// </summary>
+        /// <param name="context">The current <see cref="HttpContext"/></param>
+        /// <param name="cacheProfile">The profile which contains the cache control settings</param>
+        public void ApplyCacheProfile(HttpContext context, CacheProfile cacheProfile)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            if (cacheProfile == null)
+            {
+                throw new ArgumentNullException(nameof(cacheProfile));
+            }
+
+            var noStore = cacheProfile.NoStore ?? false;
```

Comment:
Yeah, this creates an incompatible set of arguments. Having Duration == null does not mean anything, just provide a default value. For static files I'd default to more than a minute, or even an hour.